### PR TITLE
apply security scanners results

### PR DESCRIPTION
**Pull request by hkaj**
### What does this PR do?

Update docker image to fix issues raised by image security scanners.

### Motivation

passing security scans

### Additional Notes

depends on https://github.com/DataDog/integrations-core/pull/1538 and omnibus config should be updated when it's merged.

### CODE REVIEW

**Review by CharlyF — DISMISSED** [review]
Looks good to me, just pinged Xavier for the s6 part that I'm not well versed with.

File: Dockerfiles/agent/Dockerfile [hunk 1]
```diff
@@ -83,11 +92,17 @@ COPY probe.sh initlog.sh /
 # Override the exit script by ours to fix --pid=host operations
 COPY init-stage3 /etc/s6/init/init-stage3
 
+# Extract s6-overlay
+# Done in 2 steps to avoid overriding the /bin --> /usr/bin symlink
+# FIXME: copying it from the extract stage does not work
+RUN tar xzf s6.tgz -C / --exclude="./bin" \
```
[the remark this answers is not in the document]
Comment (xvello) [reply Dockerfiles/agent/Dockerfile]: LGTM

File: Dockerfiles/agent/Dockerfile [hunk 2]
```diff
@@ -68,13 +72,18 @@ RUN if [ -n "$WITH_JMX" ]; then apt-get update \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*; fi
 
+# make sure we have recent dependencies
+RUN apt-get update && apt-get upgrade -y --no-install-recommends \
```
Comment (xvello) [inline Dockerfiles/agent/Dockerfile]: I'm afraid the `apt-get upgrade -y --no-install-recommends` might make the image size balloon up. Do we have other packages than `util-linux` and `ncurses-bin` to force-upgrade? If so, can we list them in the `apt-get install` step instead?

**Review by xvello — APPROVED** [review]
LGTM!

**Pull request merged**